Report

Flowtter opened the pull request:
Add a report option.
Useful to report an user or a message.
Available in:
- profile
- radial menu
- messages

[video](https://user-images.githubusercontent.com/34689945/195100347-dd9008cd-41ba-4a0b-aa10-3735550c1311.mp4)

Flowtter committed:
Add report

ramnes commented on core/client/ui/report.js:
```diff
@@ -0,0 +1,115 @@
+    messageText = `Message: ${messageId} - ${message.text}\nChannel: ${message.channel}\n`;
+  }
+
+  return `REPORT LOG - ${Date()}\n\n\nLevel: ${level}\n\nJustification:\n${text}\n\nReason:\n${reason}\n\n${messageText}Reported:\n${reported._id} - ${reported.profile.name}`;
```
Can't we use a multiline string rather than `\n`?

ramnes commented on core/client/ui/report.js:
```diff
@@ -0,0 +1,115 @@
+
+const getMessage = id => Messages.findOne(id);
+
+const getAllAdmins = () => Meteor.users.find({ roles: { admin: true } }).fetch();
```
these three functions don't seem necessary to me, the code already speaks for itself :)

(actually they're noise to my eyes)

ramnes commented on core/client/ui/report.js:
```diff
@@ -0,0 +1,115 @@
+  return messageId;
+};
+
+const closeConsole = () => { closeModal('report'); };
```
noise as well

ramnes commented on core/client/ui/report.js:
```diff
@@ -0,0 +1,115 @@
+  if (isAdmin) {
+    admins = admins.filter(admin => admin._id !== Meteor.userId());
+  }
+  if (!admins.length) {
+    // Quiting even if the user is an admin because we don't want to create a conversation with himself
+    lp.notif.error('No admins found');
+    return;
+  }
```
Why not disabling the button in the first place?

ramnes commented on core/client/ui/report.js:
```diff
@@ -0,0 +1,115 @@
+      case 'Other':
+        return 'Other behavior that is not listed above, please specify';
+      default:
+        return '';
```
Can't we just put this in the HTML directly?

ramnes commented on core/client/ui/report.js:
```diff
@@ -0,0 +1,115 @@
+    event.preventDefault();
+    event.stopPropagation();
+    const template = Template.instance();
+    lp.notif.confirm('Report user', `Do you really want to report this message?`, () => onSubmit(template.reason.get(), getUser(template.data.userId), template.data.messageId));
```
Does it say "this message" when we report a user through the radial menu directly?

ramnes reviewed (approved):
Just a few comments but LGTM overall 😎

xsyann committed:
Fix spawn behavior when position is not set (#181)

A user should spawn in the current level instead of the default level when
his position is not set.

Donorhan committed:
messaging: allow user to subscribe to the level channel (#179)

Flowtter committed:
Add copy paste with preview (#162)

alimtunc committed:
Add custom favicon generation + Add settings for home logo (#158)

Commit:
0 - core: update dependencies

xsyann committed:
Fix logout errors (#182)

During the logout there is a short lapse of time where Meteor.user() is null,
if the logout is performed on character or media settings screen the reactive template helpers are called and throw errors.

Donorhan committed:
3 - guilds: allow an user to edit his team from another level (#183)

Commit:
users: fix potential security issue with external link

Commit:
core: fix linter error

Flowtter committed:
Entity prefab manager (#164)

* Fix scrolling bug on the the entity template list

* Add button 'Create new prefab'

* Add edit window for the entities

* Add remove all instances button

* Add versioning on the animation

YglesEyes committed:
1 - Can disable user panel (#185)

Co-authored-by: Nicolas PILLON <nicolas@lempire.co>

Commit:
0 - core: remove reference to the missing "guestSkin" function

Commit:
meet: ensures meet is positioned to the top

Donorhan committed:
2 - notifications: stop sending webrtc notifications when the targeted zone is not the targeted channel (#186)

Commit:
1 - users: fix user lists sort errors

alimtunc committed:
Add new step onboarding for meeting people (#188)

Flowtter committed:
Fix dependencies (#189)
